lib/pure/collections/tables.nim

```diff
@@ -1232,22 +1231,23 @@ proc enlarge[A, B](t: var OrderedTable[A, B]) =
   t.last = -1
   swap(t.data, n)
   while h >= 0:
-    var nxt = n[h].next
+    let nxt = n[h].next
     let eh = n[h].hcode
-    if isFilled(eh):
-      var j: Hash = eh and maxHash(t)
-      while isFilled(t.data[j].hcode):
-        j = nextTry(j, maxHash(t))
-      rawInsert(t, t.data, move n[h].key, move n[h].val, n[h].hcode, j)
+    # every item part of the linked list is filled, so we don't need an
+    # ``isFilled`` guard here
+    var j: Hash = eh and maxHash(t)
+    # search for a free slot:
+    while isFilled(t.data[j].hcode):
+      j = nextTry(j, maxHash(t))
```

Comment:
dumb question, but it doesn't look like we're mutating `t` in the loop, so do we need to continually recalculate `maxHash(t)`?

(I'm super unfamiliar with this code, again, sorry)

Reply:
Your reading is correct, `maxHash` can be precomputed - the efficiency improvement is *likely* going to be minimal however, as `maxHash` is a template that expands to `t.data.high`. It is an improvement nonetheless, so I changed it.